app/gui/src/main/java/org/mobilitydata/gtfsvalidator/app/gui/GtfsValidatorApp.java

```diff
@@ -0,0 +1,314 @@
+    fieldConstraints.gridx = 1;
+    fieldConstraints.anchor = GridBagConstraints.NORTHEAST;
+
+    labelConstraints.gridy = 0;
+    panel.add(new JLabel("Number of threads:"), labelConstraints);
+
+    fieldConstraints.gridy = 0;
+    panel.add(numThreadsSpinner, fieldConstraints);
+    numThreadsSpinner.setValue(1);
+
+    labelConstraints.gridy = 1;
+    panel.add(new JLabel("Country Code:"), labelConstraints);
+
+    fieldConstraints.gridy = 1;
+    panel.add(countryCodeField, fieldConstraints);
+
+    advancedOptionsPanel.setVisible(false);
+  }
+
+  private void constructValidateButton(JPanel panel) {
+    JPanel validateButtonPanel = new JPanel();
+    validateButtonPanel.setLayout(new BoxLayout(validateButtonPanel, BoxLayout.LINE_AXIS));
+    panel.add(validateButtonPanel);
+
+    JButton advancedButton = new JButton("Advanced");
+    advancedButton.addActionListener(
+        (e) -> {
+          advancedOptionsPanel.setVisible(!advancedOptionsPanel.isVisible());
+          pack();
+        });
+    validateButtonPanel.add(advancedButton);
+
+    validateButtonPanel.add(Box.createHorizontalGlue());
+
+    validateButton.setText("Validate");
+    updateValidationButtonStatus();
+    validateButton.addActionListener((e) -> runValidation());
+    validateButtonPanel.add(validateButton);
+  }
+
+  private void updateValidationButtonStatus() {
+    validateButton.setEnabled(isValidationReadyToRun());
+  }
+
+  private boolean isValidationReadyToRun() {
+    if (gtfsInputField.getText().isBlank()) {
+      return false;
+    }
+    if (outputDirectoryField.getText().isBlank()) {
+      return false;
+    }
+    return true;
+  }
+
+  private void runValidation() {
+    try {
+      ValidationRunnerConfig config = buildConfig();
+      validationRunner.run(config, this);
+    } catch (Throwable ex) {
+      validationDisplay.handleError(ex);
+    }
+  }
+
+  private ValidationRunnerConfig buildConfig() throws URISyntaxException {
+    ValidationRunnerConfig.Builder config = ValidationRunnerConfig.builder();
+    config.setPrettyJson(true);
+
+    String gtfsInput = gtfsInputField.getText();
+    if (!gtfsInput.isBlank()) {
+      if (gtfsInput.startsWith("http")) {
+        config.setGtfsSource(new URI(gtfsInput));
+      } else {
+        config.setGtfsSource(Path.of(gtfsInput).toUri());
+      }
+    }
+
+    String outputDirectory = outputDirectoryField.getText();
+    if (!outputDirectory.isBlank()) {
+      config.setOutputDirectory(Path.of(outputDirectory));
+    }
+
+    Object numThreads = numThreadsSpinner.getValue();
+    if (numThreads instanceof Integer) {
+      config.setNumThreads((Integer) numThreads);
+    }
+
+    String countryCode = countryCodeField.getText();
+    if (!countryCode.isBlank()) {
+      config.setCountryCode(CountryCode.forStringOrUnknown(countryCode));
+    }
+
+    return config.build();
+  }
+
+  private static Font createBoldFont() {
+    JLabel label = new JLabel();
+    Font f = label.getFont();
```

Comment:
Nit: the purpose of the f and l variables (below) is pretty clear, so it's not really a problem, but in general maybe avoiding single letter variables would be better (except for exception and lambda)?